presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -882,7 +902,19 @@ public void dropTable(MetastoreContext metastoreContext, String databaseName, St
                     .stopOnIllegalExceptions()
                     .run("dropTable", stats.getDropTable().wrap(() ->
                             getMetastoreClientThenCall(metastoreContext, client -> {
-                                client.dropTable(databaseName, tableName, deleteData);
+                                if (!deleteFilesOnDrop) {
+                                    client.dropTable(databaseName, tableName, deleteData);
+                                }
+                                else {
+                                    Table table = client.getTable(databaseName, tableName);
+                                    client.dropTable(databaseName, tableName, deleteData);
+                                    String tableLocation = table.getSd().getLocation();
+                                    if (deleteData && isManagedTable(table) && !isNullOrEmpty(tableLocation)) {
+                                        String user = impersonationEnabled ? metastoreContext.getUsername() : DEFAULT_METASTORE_USER;
+                                        HdfsContext hdfsContext = new HdfsContext(new ConnectorIdentity(user, Optional.empty(), Optional.empty()));
```

Comment:
I checked the downstream of file deletion. Looks like the principle is not used. If that is the case, we can always use `DEFAULT_METASTORE_USER` and make this `HdfsContext` const like the one in glue metastore.

Reply:
Yes, currently it was not used, that's what my initial implementation was based on. I have reverted to make use of DEFAULT_METASTORE_USER for now. 

I am also planning to contribute another feature(Support for using Multiple S3 IAM roles based on user) which will start making use of this principle. I can make this change in that PR.